pkg/tasks/queue/pgqueue.go

```diff
@@ -541,6 +547,15 @@ func (p *PgQueue) Finish(taskId uuid.UUID, taskError error) error {
 		return ErrNotRunning
 	}
 
+	var nextRetryTime *time.Time
+	if status == config.TaskStatusFailed && info.Retries <= MaxTaskRetries {
+		upperBound := config.Get().Tasking.RetryWaitUpperBound
+		retriesRemaining := float64(MaxTaskRetries - info.Retries)
+		timeToWait := time.Minute * time.Duration(upperBound.Minutes()/(retriesRemaining+1))
+		add := time.Now().Add(timeToWait)
+		nextRetryTime = &add
```

Comment:
i set an upper bound of '10s', but it still seems to retry every 1 minute (up to 3 times), is that expected?

Reply:
It's not really expected, but I wrote this expecting to use hours (or 0 for tests). Let me see if I can adjust this to be a little more flexible.

Reply:
I'm seeing a little something funky.  When i set the RetryWaitUpperBound to 10s, it sets the nextRetryTime to essentially time.Now(), which i assume means that add is basically zero.  (I confirmed the nextRetryTime by looking in the db).  But the task doesn't actually execute until about 30 seconds later.

I set it to 1m, and the nextRetryTime did get set in the future, but the task didn't actually execute until 20 seconds or so after that time passed.  I this expected?

Reply:
What is your heartbeat timer set to? The tasks will get requeued once per heartbeat. Could that be why you're seeing a delay?

Reply:
ahhh!, that make sense.  Does the other  behavior sound fine?  as you said i don't think we'd actually use these low value, so i think its okay to me?

Reply:
I guess it is a bit confusing. But since these low values are only ever used for testing purposes, I think it's okay. It's simpler this way.